01 SFC Hello VueJS/README.md

```diff
@@ -0,0 +1,264 @@
+- Configure `webpack.config` to set `vue.js` as vendor:
+
+### ./webpack.config.js
+
+```diff
+...
+  entry: {
+    app: './main.ts',
++   vendor: [
++     'vue',
++   ],
+    vendorStyles: [
+      '../node_modules/bootstrap/dist/css/bootstrap.css',
+    ],
+  },
+...
+```
+
+- Recommended configuration for [`tsconfig.json`](https://vuejs.org/v2/guide/typescript.html#Recommended-Configuration):
+
+### ./tsconfig.json
+```diff
+{
+  "compilerOptions": {
+    "target": "es6",
+    "module": "es6",
+    "moduleResolution": "node",
+    "declaration": false,
+    "noImplicitAny": false,
+    "sourceMap": true,
+    "noLib": false,
+-   "suppressImplicitAnyIndexErrors": true
++   "suppressImplicitAnyIndexErrors": true,
++   "strict": true
+  },
+  "compileOnSave": false,
+  "exclude": [
+    "node_modules"
+  ]
+}
+
+```
+
+- There are [two builds available](https://vuejs.org/v2/guide/installation.html#Explanation-of-Different-Builds) to work with `Vue.js`.
+
+  - `Runtime + Compiler`: needed to compile templates on the client side, e.g. passing a string to `template` property.
+  - `Runtime-only`: when using a `pre-compiler` like `vue-loader`, `vueify`, etc.
+
+- For now, we could start using `first one` to keep this sample as simple as possible. To configure it for webpack:
+
+### ./webpack.config.js
+
+```diff
+...
+resolve: {
+-    extensions: ['.js', '.ts'],
++    extensions: ['.js', '.ts', 'vue'],
++    alias: {
++      'vue$': 'vue/dist/vue.esm.js'
++    },
+  },
+...
+  module: {
+    rules: [
++     {
++       test: /\.vue$/,
++       exclude: /node_modules/,
++       loader: 'vue-loader'
++     },
+      {
+        test: /\.ts$/,
+        exclude: /node_modules/,
+        use: {
+          loader: 'ts-loader',
+          options: {
++           appendTsSuffixTo: [/\.vue$/],
+            transpileOnly: true,
+          },
+        },
+      },
+···
+  plugins: [
++   new VueLoaderPlugin(),
+···
+    new ForkTsCheckerWebpackPlugin({
+      tsconfig: path.join(__dirname, './tsconfig.json'),
++     vue: true
+    }),
+  ],
+};
+
+```
+
+- Update `tsconfig.json`:
+
+```diff
+{
+···
+  "compileOnSave": false,
++  "include": [
++    "src/**/*.ts",
++    "src/**/*.vue"
++  ],
+  "exclude": [
+    "node_modules"
+  ]
+}
+
+```
+
+- Update `index.html`:
+
+### ./src/index.html
+
+```diff
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>Vue.js by sample</title>
+  </head>
+  <body>
+    <div class="well">
+      <h1>Sample app</h1>
+    </div>
++   <div id="root">
++   </div>
+  </body>
+</html>
+```
+
+- Create `App.vue`:
+
+### ./src/App.vue
+
+```javascript
+<template>
+  <h1>{{ message }}</h1>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+
+export default Vue.extend({
+  name: 'App',
+  data() {
+    return {
+      message: 'Hello from Vue.js',
+    };
+  },  
+});
+</script>
+
+```
```

Comment:
Important install `[vetur VS Code plugin](https://github.com/vuejs/vetur)` to get intellisense in .vue files

Reply:
Perfect.